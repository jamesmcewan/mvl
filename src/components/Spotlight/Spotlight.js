import React from 'react';
import styled, { css } from 'styled-components';
import Detail from '../Detail/Detail';

const SpotlightSection = styled.section`
  background: #151617 linear-gradient(#1a0f21, #151617 70%) no-repeat;
  position: relative;
  width: 100%;
  box-sizing: border-box;
  margin-bottom: 52px;
  display: none;
  order: 1;

  @media screen and (min-width: 700px) {
    padding: 0;
  }

  ${props =>
    props.isSpotlightVisible &&
    css`
      display: block;
      @media screen and (min-width: 1000px) {
        display: flex;
        align-items: flex-start;
      }
    `};
`;

const SpotlightImg = styled.img`
  padding: 0;
  margin-bottom: 26px;
  width: 100%;
  box-sizing: border-box;
  border: 0;
  background: transparent;
  height: 200px;
  object-fit: cover;
  object-position: left top;

  @media screen and (min-width: 700px) {
    height: 400px;
  }

  @media screen and (min-width: 1000px) {
    width: 450px;
    height: auto;
  }
`;

const Spotlight = ({
  title,
  description,
  thumbnail,
  close,
  creators,
  diamondCode,
  urls,
  isSpotlightVisible,
}) => {
  const { path, extension } = thumbnail;
  const detailProps = {
    title,
    description,
    creators,
    diamondCode,
    urls,
    close,
  };

  return (
    <SpotlightSection isSpotlightVisible={isSpotlightVisible}>
      <SpotlightImg src={`${path}.${extension}`} alt={title} />
      <Detail {...detailProps} />
    </SpotlightSection>
  );
};

Spotlight.defaultProps = {
  title: '',
  description: '',
  creators: [
    {
      name: '',
      role: '',
    },
  ],
  dates: [
    {
      date: '',
      type: '',
    },
  ],
  urls: [
    {
      url: '',
      type: '',
    },
  ],
  thumbnail: {
    path: '',
    extension: '',
  },
};

export default Spotlight;
