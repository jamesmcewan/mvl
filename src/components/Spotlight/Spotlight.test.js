import React from 'react';
import { mount } from 'enzyme';
import Spotlight from './Spotlight';
import 'jest-styled-components';

describe('Spotlight', () => {
  beforeEach(() => {
    window.scrollTo = jest.fn();
  });

  it('should render without crashing', () => {
    const wrapper = mount(<Spotlight />);
    expect(wrapper).toHaveLength(1);
  });

  it('should render the correct JSX', () => {
    const wrapper = mount(<Spotlight />);
    expect(wrapper).toMatchSnapshot();
  });
});
