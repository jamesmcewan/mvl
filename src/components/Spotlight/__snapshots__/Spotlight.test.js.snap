// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spotlight should render the correct JSX 1`] = `
.c6 {
  list-style-type: none;
  margin: 0;
  padding: 0;
}

.c5 {
  margin-top: 0;
  font-size: 22px;
}

.c2 {
  padding-bottom: 26px;
  margin: 0 13px;
}

.c3 {
  font-family: 'Roboto Slab',serif;
  margin-top: 0;
}

.c4 {
  margin-bottom: 26px;
}

.c8 {
  border: 1px solid #ff79c6;
  background: transparent;
  font-size: 18px;
  padding: 9px;
  border-radius: 10px;
  color: #ff79c6;
}

.c0 {
  background: #151617 linear-gradient(#1a0f21,#151617 70%) no-repeat;
  position: relative;
  width: 100%;
  box-sizing: border-box;
  margin-bottom: 52px;
  display: none;
  -webkit-order: 1;
  -ms-flex-order: 1;
  order: 1;
}

.c1 {
  padding: 0;
  margin-bottom: 26px;
  width: 100%;
  box-sizing: border-box;
  border: 0;
  background: transparent;
  height: 200px;
  object-fit: cover;
  object-position: left top;
}

@media screen and (min-width:700px) {
  .c2 {
    padding: 0;
    margin: 0 6px 6px;
  }
}

@media screen and (min-width:1000px) {
  .c2 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-flow: row wrap;
    -ms-flex-flow: row wrap;
    flex-flow: row wrap;
    -webkit-align-items: flex-start;
    -webkit-box-align: flex-start;
    -ms-flex-align: flex-start;
    align-items: flex-start;
    margin-left: 52px;
  }
}

@media screen and (min-width:1000px) {
  .c3 {
    -webkit-flex: 1 100%;
    -ms-flex: 1 100%;
    flex: 1 100%;
  }
}

@media screen and (min-width:1000px) {
  .c7 {
    -webkit-flex: 1 100%;
    -ms-flex: 1 100%;
    flex: 1 100%;
  }
}

@media screen and (min-width:1000px) {
  .c4 {
    max-width: 600px;
    margin-right: 52px;
  }
}

@media screen and (min-width:700px) {
  .c0 {
    padding: 0;
  }
}

@media screen and (min-width:700px) {
  .c1 {
    height: 400px;
  }
}

@media screen and (min-width:1000px) {
  .c1 {
    width: 450px;
    height: auto;
  }
}

<Spotlight
  creators={
    Array [
      Object {
        "name": "",
        "role": "",
      },
    ]
  }
  dates={
    Array [
      Object {
        "date": "",
        "type": "",
      },
    ]
  }
  description=""
  thumbnail={
    Object {
      "extension": "",
      "path": "",
    }
  }
  title=""
  urls={
    Array [
      Object {
        "type": "",
        "url": "",
      },
    ]
  }
>
  <styled.section>
    <section
      className="c0"
    >
      <styled.img
        alt=""
        src="."
      >
        <img
          alt=""
          className="c1"
          src="."
        />
      </styled.img>
      <Detail
        creators={
          Array [
            Object {
              "name": "",
              "role": "",
            },
          ]
        }
        dates={
          Array [
            Object {
              "date": "",
              "type": "",
            },
          ]
        }
        description=""
        title=""
        urls={
          Array [
            Object {
              "type": "",
              "url": "",
            },
          ]
        }
      >
        <styled.div>
          <div
            className="c2"
          >
            <styled.h2>
              <h2
                className="c3"
              />
            </styled.h2>
            <styled.div
              dangerouslySetInnerHTML={
                Object {
                  "__html": "",
                }
              }
            >
              <div
                className="c4"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "",
                  }
                }
              />
            </styled.div>
            <Creators
              creators={
                Array [
                  Object {
                    "name": "",
                    "role": "",
                  },
                ]
              }
            >
              <div>
                <styled.h3>
                  <h3
                    className="c5"
                  >
                    Creative team
                  </h3>
                </styled.h3>
                <styled.ul>
                  <ul
                    className="c6"
                  >
                    <Creator
                      key="c-"
                      name=""
                      role=""
                    >
                      <li>
                         - 
                      </li>
                    </Creator>
                  </ul>
                </styled.ul>
              </div>
            </Creators>
            <styled.p>
              <p
                className="c7"
              />
            </styled.p>
            <styled.button
              onClick={[Function]}
            >
              <button
                className="c8"
                onClick={[Function]}
              >
                Close
              </button>
            </styled.button>
          </div>
        </styled.div>
      </Detail>
    </section>
  </styled.section>
</Spotlight>
`;
